### PR TITLE
add cosign pkg

### PR DESCRIPTION
**Pull request by tuananh**
Signed-off-by: Tuan Anh Tran <me@tuananh.org>

### CODE REVIEW

File: cosign.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  description: Container Signing
+  target-architecture:
+    - all
+  copyright:
+    - license: Apache-2.0
+      paths:
+        - "*"
+  dependencies:
+    runtime:
+      - apk-tools
```
Comment (imjasonh) [inline cosign.yaml:13]: We shouldn't need this at runtime. I think we will need CA certs at runtime though.

File: cosign.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package:
+  name: cosign
+  version: 1.13.1
+  description: Container Signing
+  target-architecture:
```
Comment (kaniini) [inline cosign.yaml:5]: Please add `epoch: 0` here, even if it is moot, it makes it easier to bump the epoch later if it is already present.

File: cosign.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      paths:
+        - "*"
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+
+environment:
+  contents:
+    keyring:
+      - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
```
Comment (kaniini) [inline cosign.yaml:18]: `keyring`/`repositories` are redundant here.

**Review by kaniini — APPROVED** [review]

File: cosign.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - go
+
+pipeline:
+  - uses: fetch
+    with:
+      uri: https://github.com/sigstore/cosign/archive/v${{package.version}}/cosign-v${{package.version}}.tar.gz
+      expected-sha256: 496583e9a3e07f16ff18f3734ca69398ef7091336fe74adaf8012715656bb920
+  - runs: |
+      CGO_ENABLED=0 go build \
+        -o "${{targets.destdir}}/usr/bin/cosign" ./
```
Comment (kaniini) [inline cosign.yaml:33]: probably needs to be `go build ./...`.

**Review by kaniini — CHANGES_REQUESTED** [review]

File: cosign.yaml (hunk 4 above)
Comment (imjasonh) [reply cosign.yaml:33]: I think this should be `go build ./cmd/cosign`

Comment (cpanato) [reply cosign.yaml:33]: Can we also inject the version in the ldflags?

Comment (tuananh) [reply cosign.yaml]: i injected the version but git state is not there in the release tar archive so I just leave it out.

File: cosign.yaml (hunk 3 above)
Comment (tuananh) [reply cosign.yaml:19]: i dont need to use keyring if repository is there? i saw the other packages they all have both.

Comment (kaniini) [reply cosign.yaml:19]: Some do but it is a problem for a few reasons so trying to avoid it

File: cosign.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    - license: Apache-2.0
+      paths:
+        - "*"
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+
+environment:
+  contents:
+    repositories:
```
Comment (kaniini) [inline cosign.yaml:18]: The `repositories` entry is redundant.

**Comment by tuananh** [comment]
@imjasonh can you help with debugging the new lint script

i got error with this but everything looks good to me

```
--- cosign.yaml
build-package,cosign
,1.13.1
-r0
 not found in Makefile
Error: Process completed with exit code 1.
```

**Comment by tuananh** [comment]
@kaniini can you help? i'm not sure what i did that make the lint script failed here.

**Comment by rawlingsj** [comment]
The issue is with the lineendings in the `cosign.yaml` file.  I can push a PR to your branch if you like but it might be worth addressing yourself so future PRs don't suffer from the same issue.

I was able to recreate the problem locally by running the lint script locally but I'm not 100% sure if that will fail for you, are you on windows maybe?
```
./lint.sh
```

When I open your fork and cosign branch in VSCode, in the bottom right hand corner I see `CRLF`

Screenshot 2022-11-16 at 10 09 14

If you compare that to other yamls in that folder they have `LF`

In VSCode you can click that `CRLF` in the bottom bar and then select `LF`

Screenshot 2022-11-16 at 10 11 49

Now rerunning the `./lint.sh` script should pass.

See if that helps?  Also you'll need to do a quick rebase for the makefile conflict but that should be a small diff.

**Comment by tuananh** [comment]
> 

this is it. thank you :)

**Comment by tuananh** [comment]
is there anything else you may want to change @kaniini ?

**Review by kaniini — APPROVED** [review]

**Pull request merged**